### PR TITLE
Feature/activity backend

### PR DESCRIPTION
**Pull request by ndeasy**
# Create endpoints for creating and retrieving activities

Tests being done as separate subtask

Changed the activity service to accommodate the latest UI changes. 

Can now create an activity, and retrieve a list of your created activities. Requires a valid authentication session. 

Endpoint: /api/activity

Create [**POST**] a new activity from the posted json
Delete(id) [**Delete**] activity who's id matches the parameter passed
Get [**GET**] all of the activities created by that user. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Updated tests

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: activity-recorder/src/routes/activity.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    '/api/activity',
+    validateRequest,
+    async (req: Request, res: Response) => {
+
+        await Activity.deleteOne({id: req.body.id})
+        res.status(200).send("deleted activity");
+    },
+)
+
+router.get('/api/activity', currentUser, async (req: Request, res: Response) => {
```
Comment (hb308) [inline activity-recorder/src/routes/activity.ts:18]: Do you want to maybe separate the routers in different files?

File: activity-recorder/src/routes/activity.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    const usersActivities = await Activity.find({ userId: req.currentUser!.id });
+    res.status(200).send(usersActivities);
+});
+
+router.post(
```
Comment (hb308) [inline activity-recorder/src/routes/activity.ts:23]: Once you've got the login stuff merged, you can add in the requireAuth middleware :)

File: activity-recorder/src/routes/activity.ts (hunk 1 above)
Comment (ndeasy) [reply activity-recorder/src/routes/activity.ts:18]: it's the same route just different request types, I think they should be in one file.

File: activity-recorder/src/routes/activity.ts (hunk 2 above)
Comment (ndeasy) [reply activity-recorder/src/routes/activity.ts]: done

File: activity-recorder/src/routes/log.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import express, { Request, Response } from 'express';
+import { validateRequest } from '@amp-rehab-app/common';
+
+const router = express.Router();
+
+router.post(
```
Comment (hb308) [inline activity-recorder/src/routes/log.ts:6]: Are we using this or can it be removed?

Comment (ndeasy) [reply activity-recorder/src/routes/log.ts:6]: it can be removed, but I thought I would add a template for logging activities which is the next ticket

Comment (hb308) [reply activity-recorder/src/routes/log.ts:6]: Okay thats perfect, you can leave it as that :)

**Review by hb308 — APPROVED** [review]
Looks good to merge :)

**Pull request merged**